tests/fixture/issue-1555/output/index.js

```diff
@@ -1,5 +1,5 @@
 export default function _setPrototypeOf(o, p) {
-    _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o1, p1) {
+    _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf1(o1, p1) {
```

Comment:
I found that this is a new bug.
It should be `_setPrototypeOf` I think.

Reply:
I think this is the same case as described above. The inner function name is a separate binding from the outer one which is re-assigned. Assigning to the outer one doesn't re-assign the inner one. If the inner function was referenced, it would still refer to the inner function even if the outer binding is re-assigned.

I think this is correct, though again, perhaps the hygiene pass could avoid renaming in cases like this where a binding is shadowed in an inner scope and the outer binding is not referenced within that scope. That would also improve the output for cases like:

```js
let x = 2;
{
  let x = 3;
}
```

which currently results in:

```js
let x = 2;
{
    let x1 = 3;
}
```

The resolver is correct - the inner x is a different binding from the outer one, but it didn't really need to be renamed to `x1` since the outer binding is not referenced in the inner scope (or any of its children).

Reply:
Here's an example:

```js
let test = function test() {
  console.log(test);
};

setTimeout(test, 0);
test = 2;
console.log(test);
```

If you run that, you'll get:

```
2
[Function: test]
```

Within the function, `test` always refers to the function, not the outer variable. If you change the code to:

```js
let test = function() {
  console.log(test);
};

setTimeout(test, 0);
test = 2;
console.log(test);
```

Then you'll get `2` twice since there's no inner binding.

Reply:
Sorry. I misread it. (I was in hurry because of the midterm exam and now it's over)

Yes, it's the same as the case above.

Reply:
And thank you for the example :)